Mathlib/Algebra/Algebra/Subalgebra/Unitization.lean

```diff
@@ -258,33 +295,81 @@ theorem StarSubalgebra.toNonUnitalStarSubalgebra_toStarSubalgebra (S : StarSubal
 
 theorem NonUnitalStarSubalgebra.toStarSubalgebra_toNonUnitalStarSubalgebra
     (S : NonUnitalStarSubalgebra R A) (h1 : (1 : A) ∈ S) :
-    (NonUnitalStarSubalgebra.toStarSubalgebra S h1).toNonUnitalStarSubalgebra = S := by
+    (S.toStarSubalgebra h1).toNonUnitalStarSubalgebra = S := by
   cases S; rfl
 
+section StarLiftRange
+
+variable {R A C : Type*} [CommSemiring R] [NonUnitalSemiring A] [StarRing R] [StarRing A]
+variable [Module R A] [SMulCommClass R A A] [IsScalarTower R A A] [StarModule R A]
+variable [Semiring C] [StarRing C] [Algebra R C] [StarModule R C]
+
+theorem Unitization.starLift_range_le
+    {f : A →⋆ₙₐ[R] C} {S : StarSubalgebra R C} :
+    (starLift f).range ≤ S ↔ NonUnitalStarAlgHom.range f ≤ S.toNonUnitalStarSubalgebra := by
+  refine ⟨fun h ↦ ?_, fun h ↦ ?_⟩
+  · rintro - ⟨x, rfl⟩
+    exact @h (f x) ⟨x, by simp⟩
+  · rintro - ⟨x, rfl⟩
+    induction x using Unitization.ind with
+    | _ r a => simpa using add_mem (algebraMap_mem S r) (h ⟨a, rfl⟩)
+
+theorem Unitization.starLift_range (f : A →⋆ₙₐ[R] C) :
+    (starLift f).range = StarSubalgebra.adjoin R (NonUnitalStarAlgHom.range f : Set C) :=
+  eq_of_forall_ge_iff fun c ↦ by
+    rw [Unitization.starLift_range_le, StarSubalgebra.adjoin_le_iff]
+    rfl
+
+end StarLiftRange
+
 /-- The natural star `R`-algebra homomorphism from the unitization of a non-unital star subalgebra
 to its `StarSubalgebra.adjoin`. -/
 def NonUnitalStarSubalgebra.unitization (S : NonUnitalStarSubalgebra R A) :
```

Comment:
Maybe we discussed that already, but why can't you use `SetLike` here like you did for subalgebras? The same question applies to the subring and subsemiring results.

Reply:
good point, I've transitioned to `SetLike` everywhere